src/modules/requirements/MegaEvolveRequirement.ts

```diff
@@ -1,16 +1,29 @@
-import { AchievementOption } from '../GameConstants';
+import { AchievementOption, MegaStoneType } from '../GameConstants';
+import { pokemonMap } from '../pokemons/PokemonList';
 import { PokemonNameType } from '../pokemons/PokemonNameType';
 import Requirement from './Requirement';
 
 export default class MegaEvolveRequirement extends Requirement {
-    constructor(private name: PokemonNameType) {
+    private readonly baseAttackMultiplier = 500;
+
+    constructor(private name: PokemonNameType, private megaStone: MegaStoneType) {
         super(1, AchievementOption.equal);
     }
 
     getProgress(): number {
-        return App.game.party.getPokemonByName(this.name)?.megaStone?.canEvolve() ? 1 : 0;
+        const partyPokemon = App.game.party.getPokemonByName(this.name);
+        return player.itemList[MegaStoneType[this.megaStone]]() > 0
+            && partyPokemon?.attack >= pokemonMap[this.name].attack * this.baseAttackMultiplier ? 1 : 0;
     }
+
     hint(): string {
-        return App.game.party.getPokemonByName(this.name)?.megaStone?.getTooltipText() || `${this.name} holds no Mega Stone.`;
+        const attackRequired = pokemonMap[this.name].attack * this.baseAttackMultiplier;
+        if (this.getProgress()) {
+            return 'Use a Key Stone to Mega Evolve.';
+        } else if (App.game.party.getPokemonByName(this.name).attack < attackRequired) {
+            return `Needs at least ${attackRequired.toLocaleString('en-US')} attack to Mega Evolve.`;
+        } else {
+            return `${this.name} holds no Mega Stone.`;
```

Comment:
It's probably better to name the mega stone needed here? Something like
```ts
`${this.name} needs the ${MegaStoneType[this.megaStone]} Mega Stone.`
```

Also I think it should either be shown at the same time as the attack requirement, or override it when the player doesn't have the stone yet.

Reply:
Updated